### PR TITLE
feat: bump reference-implementation version and fix errors

### PR DESCRIPTION
**Pull request by jaypaik**
Relevant updates:
- Bump dependency versions:
  - `"@erc6900/modular-account-libs": "github:erc6900/modular-account-libs#v0.8.0-rc.1"`
  - `"@erc6900/reference-implementation": "github:erc6900/reference-implementation#v0.8.0-rc.3"`
- `executeWithAuthorization` -> `executeWithRuntimeValidation`
- AccountStorage updated to use `HookConfig[] validationHooks` instead of `ModuleEntity[] preValidationHooks`
  - Question here: should we just keep storage to `ModuleEntity[]` and do the transformation to `HookConfig[]` in the view function instead?

### CODE REVIEW

File: test/mocks/modules/ReturnDataModuleMocks.sol [hunk 1]
```diff
@@ -101,9 +101,9 @@ contract ResultConsumerModule is IExecutionModule, BaseModule, IValidationModule
     }
 
     // Check the return data through the execute with authorization case
-    function checkResultExecuteWithAuthorization(address target, bytes32 expected) external returns (bool) {
+    function checkResultexecuteWithRuntimeValidation(address target, bytes32 expected) external returns (bool) {
```
Comment (adamegyed) [inline test/mocks/modules/ReturnDataModuleMocks.sol]:
```suggestion
    function checkResultExecuteWithRuntimeValidation(address target, bytes32 expected) external returns (bool) {
```

File: test/mocks/modules/ReturnDataModuleMocks.sol [hunk 2]
```diff
@@ -129,7 +129,7 @@ contract ResultConsumerModule is IExecutionModule, BaseModule, IValidationModule
             allowGlobalValidation: false
         });
         manifest.executionFunctions[1] = ManifestExecutionFunction({
-            executionSelector: this.checkResultExecuteWithAuthorization.selector,
+            executionSelector: this.checkResultexecuteWithRuntimeValidation.selector,
```
Comment (adamegyed) [inline test/mocks/modules/ReturnDataModuleMocks.sol]:
```suggestion
            executionSelector: this.checkResultExecuteWithRuntimeValidation.selector,
```

**Review by adamegyed — APPROVED** [review]
Looks good, 1 small capitalization nit.

> should we just keep storage to ModuleEntity[] and do the transformation to HookConfig[] in the view function instead?

I'm OK with either, but I have a slight suspicion that the version with`HookConfig` is what's causing the marginal increase in gas costs. We can change it if we find it improves performance.

File: test/mocks/modules/ReturnDataModuleMocks.sol (hunk 1 above)
Comment (jaypaik) [reply test/mocks/modules/ReturnDataModuleMocks.sol]: Great catches!

**Comment by jaypaik** [comment]
> I'm OK with either, but I have a slight suspicion that the version withHookConfig is what's causing the marginal increase in gas costs. We can change it if we find it improves performance.

Tried this and actually seems cheaper on average to store HookConfig. Reverting back.

**Pull request merged**